login.go

```diff
@@ -73,20 +73,25 @@ func (p *Params) Login() (connection interface{}, loginInfo LoginInfo, err error
 			}
 		}()
 	}
-
 	defer close(p.doneChan)
-	//for credential map in slice
-	for _, credentials := range p.Credentials {
+	//for each IpAddress
+	for _, ip := range p.IpAddresses {
+		if ip == "" {
+			continue
+		}
 
-		//for each credential k, v
-		for user, pass := range credentials {
+		hostreachable, _ := p.IsHostReachable(ip, 443)
+		if !hostreachable {
+			log.Printf("Host %s is not reachable at port 443", ip)
+			continue
+		}
 
-			//for each IpAddress
-			for _, ip := range p.IpAddresses {
-				if ip == "" {
-					continue
-				}
+		//for credential map in slice
+		for _, credentials := range p.Credentials {
 
+			//for each credential k, v
+			for user, pass := range credentials {
+				backoff = 0
```

Comment:
As a suggestion for a future change, switch to https://pkg.go.dev/github.com/cenkalti/backoff/v4 for dealing with backoffs.

Reply:
Will keep in mind. Thanks @joelrebel